jenkins/Jenkinsfile

```diff
@@ -160,30 +139,35 @@ EOT
         stage('build_rpm'){
           steps {
             container('rpmbuild'){
-              sh script: """
+              sh script: '''
                 cd ~/rpmbuild
-                git clone https://github.com/cloudify-cosmo/cloudify-agent.git SOURCES && cd SOURCES && git checkout ${env.BRANCH_NAME}
-              """, label: 'Copying repo to separate workspace'
+                git clone https://github.com/cloudify-cosmo/cloudify-agent.git SOURCES && cd SOURCES && git checkout ${BRANCH_NAME}
+              ''', label: 'Copying repo to separate workspace'
               sh script: """
                 curl -o ~/rpmbuild/SOURCES/cloudify-agents.spec https://raw.githubusercontent.com/cloudify-cosmo/cloudify-manager/master/packaging/cloudify-agents.spec ;
                 mkdir -p ~/rpmbuild/SOURCES/packaging/agents/ ;
                 curl -o ~/rpmbuild/SOURCES/packaging/agents/copy_packages.py https://raw.githubusercontent.com/cloudify-cosmo/cloudify-manager/master/packaging/agents/copy_packages.py ;
                 """, label: 'Prepare rpm-building environment'
-              sh script: """
+              sh script: '''
                 cd ~/rpmbuild/SOURCES
-                cp -rf ${env.WORKSPACE}/build_agent/*.tar.gz .
+
+                #--- build the agent here
+                cp /rpmbuild/SOURCES/packaging/linux_agent_builder.sh .
+                chmod +x linux_agent_builder.sh
+                ./linux_agent_builder.sh centos core ${VERSION}-${PRERELEASE}
+
                 for i in {1..30}; do yum-builddep -y cloudify-agents.spec && break || sleep 10; done
```

Comment:
While we're here can we add a comment noting that we do this up to 30 times because yum mirrors aren't always entirely reliable?
I think I put a bunch of these retries in various places, so should've commented them earlier, sorry!